src/ast/passes/semantic_analyser.cpp

```diff
@@ -514,7 +519,39 @@ void SemanticAnalyser::visit(Call &call)
       auto &expr = (*call.vargs)[i];
       func_arg_idx_ = i;
 
-      expr->accept(*this);
+      // Handling offsetof() 2nd argument
+      // Handle the second parameter offsetof(record, field), this parameter
+      // is treated as an Identifier, of course, it may be call or builtin, we
+      // convert them all to Identifiers. For example:
+      //
+      //  struct foo {
+      //    int offsetof;   > offsetof(struct foo, offsetof)
+      //    int kstack;     > offsetof(struct foo, kstack)
+      //    int ctx;        > offsetof(struct foo, ctx)
+      //    int comm;       > offsetof(struct foo, comm)
+      //  }
+      if (func_ == "offsetof" && func_arg_idx_ == 1)
+      {
+        auto _builtin = dynamic_cast<Builtin *>(call.vargs->at(1));
+        auto _call = dynamic_cast<Call *>(call.vargs->at(1));
+
+        if (_builtin)
+        {
+          delete expr;
+          expr = new Identifier(_builtin->ident, call.loc);
+        }
+        else if (_call)
+        {
+          delete expr;
+          expr = new Identifier(_call->func, call.loc);
+        }
```

Comment:
```suggestion
        }
        auto _ident = dynamic_cast<Identifier *>(call.vargs->at(1));
        if (!_ident)
        {
          LOG(ERROR) << "...";
        }
```
Not sure if that can happen, though.

Reply:
Unfortunately, this will not happen, because it will be processed as `assert(ident)` and will be printed directly:

```
Assertion `element && "offsetof() 2nd argument not an identifier"' failed.
```